src/main/java/com/google/api/codegen/viewmodel/OutputView.java

```diff
@@ -0,0 +1,85 @@
+/* Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.viewmodel;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+
+@AutoValue
+public abstract class OutputView {
+
+  public enum Kind {
+    PRINT
+  }
+
+  public abstract Kind kind();
+
+  public abstract PrintView print();
+
+  public static Builder newBuilder() {
+    return new AutoValue_OutputView.Builder();
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+    public abstract Builder kind(Kind val);
+
+    public abstract Builder print(PrintView val);
+
+    public abstract OutputView build();
+  }
+
+  @AutoValue
+  public abstract static class PrintView {
+
+    public abstract String printSpec();
```

Comment:
What if instead we only had `args`, where `args[0]` is the format string if `len(args)>1`? This makes it simpler in the .snip to concatenate all the arguments without having the empty `{}` format string in your current baseline.

In this case, if `len(args)==0`, the transformer could set `args[0]` to `"\n"`

(This ties in nicely, I think, with my other comment below.)

Reply:
Synced offline. We shouldn't do this since langs like Python does `"formatspec".format(...)`